enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,211 @@
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "Metal3ProvisioningController" would be created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCP : This configuration consists of two components:
+	Management type
+	DHCP range
+If the DHCP type is "Removed" it means that the metal3 cluster would not be
+responsible for managing DHCP addresses and an external DHCP server is expected to be
+available and reachable by the cluster.
+
+If the Management type is "Managed", then the DHCP range indicates the pool of
+IP addresses that can used to assign to the baremetal hosts. If the range is unset, then
+the default IP address range (.10 to .100) would be used. The value of the DHCP range
+can be changed even after installation but the value of the Management type cannot be
+changed after installation. 
+
+### User Stories [optional]
+
+1. As a Deployment Operator, I want Barametal IPI deployments to be customizable to
+hardware and network requirements.
+
+2. As an Openshift Administrator, I want Baremetal IPI deployments to take place without
+manual workarounds like creating a ConfigMap for the config (which is the current approach
+being used in 4.2 and 4.3.) 
+
+## Design Details
+
+This new baremetal CR would be created in the "openshift-machine-api" namespace and as
+mentioned earlier would be in operator scope. It will belong to the group
+"baremetal.operator.openshift.io". This would be a singleton CR
+and would have the name "metal3provisioningcontroller". This new CR would be implemented
+with an API version of apiextensions.k8s.io/v1.
+
+In summary,
+Resource name - baremetal.operator.openshift.io
+Instance name - metal3provisioningcontroller (? Not sure)
```

Comment:
this makes no sense to me, any instance ever would be  a `provisioning` config, this should not belong to the specific instance name but to the resource name. e.g
resource: `provisioningnetwork.baremetal.operator.openshift.io`, instance: `cluster`/`default`

cc @bison @abhinavdahiya

Reply:
also I'm know if was discarded in previous proposal but just so I'm aware why is this prefered over a just a configmap?

Reply:
My understanding is that a ConfigMap is not appropriate because we cannot version it, which introduces concerns about upgrading.

Reply:
That approach was rejected (over 4 months ago) in https://github.com/openshift/installer/pull/2149

It's pretty frustrating that we're still here failing to reach consensus on how to wire in a few key/value pairs of config data after so much time.

Reply:
Just wanted to get more context @hardys. I hear you and I agree, this pretty much lgtm let's try to close any pending comments and move forward

Reply:
I tried to reverse engineer the Resource name and Instance Name by looking at other CRDs in openshift/api. I have updated the doc based on comments in this thread.